### PR TITLE
Misra: Rule 10.4 violations

### PR DESCRIPTION
**Pull request by alfred2g**
Misra: Rule 10.4 violations

Description
-----------
Fix misra Rule 10.4 violations.

Test Steps
-----------
Violations disappear form the misra checker

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/FreeRTOS_DNS_Cache.c [hunk 1]
```diff
@@ -340,26 +340,26 @@
         /* Add or update the item. */
         if( strlen( pcName ) < ( size_t ) ipconfigDNS_CACHE_NAME_LENGTH )
```
Comment (paulbartell) [inline source/FreeRTOS_DNS_Cache.c:341]:
```suggestion
        if( strnlen( pcName, ipconfigDNS_CACHE_NAME_LENGTH ) < ( size_t ) ipconfigDNS_CACHE_NAME_LENGTH )
```

**Review by paulbartell — APPROVED** [review]

File: source/FreeRTOS_Sockets.c [hunk 2]
```diff
@@ -4112,8 +4112,8 @@ void vSocketWakeUpUser( FreeRTOS_Socket_t * pxSocket )
                 { 25U, 100U }, /* 4 segments. */
             };
 
-            if( ( uxSegmentCount > 0 ) &&
-                ( uxSegmentCount <= ARRAY_SIZE( xPercTable ) ) )
+            if( ( uxSegmentCount > 0U ) &&
+                ( uxSegmentCount <= ( UBaseType_t ) ARRAY_SIZE( xPercTable ) ) )
```
Comment (htibosch) [inline source/FreeRTOS_Sockets.c:4116]: `uxSegmentCount` is of type `size_t`, not `UBaseType_t`.

File: source/FreeRTOS_DNS_Cache.c [hunk 3]
```diff
@@ -72,7 +72,7 @@
  * @brief indicates the index of a free entry in the cache structure
  *        \a  DNSCacheRow_t
  */
-    static BaseType_t xFreeEntry = 0;
+    static UBaseType_t uxFreeEntry = 0;
```
Comment (htibosch) [inline source/FreeRTOS_DNS_Cache.c:75]: The constant value also wants to be unsigned `0U`

File: source/FreeRTOS_DNS_Cache.c [hunk 4]
```diff
@@ -139,7 +139,7 @@
     void FreeRTOS_dnsclear( void )
     {
         ( void ) memset( xDNSCache, 0x0, sizeof( xDNSCache ) );
-        xFreeEntry = 0;
+        uxFreeEntry = 0;
```
Comment (htibosch) [inline source/FreeRTOS_DNS_Cache.c:142]: Change it to `0U` ?

**Review by htibosch — APPROVED** [review]
Thank you @alfred2g for these changes. Beside a few minor suggestions I approve it.

**Pull request merged**